dianatanase/assignment1/Q2.java

```diff
@@ -0,0 +1,137 @@
+import java.util.Scanner;
+
+// Note: I could have used a LinkedList<T> with its size() and get() method
+// but I have created my SimpleLinkedList and Node objects for practice.
+
+// Since the type of objects in the list is not known
+// Create a Node object with data of Generic type T to represent the nodes in the list.
+// A Node has data (value) and is linked to a next Node (can be null).
+class Node<T>{
+	
+	T data = null;
+	Node<T> next = null;
+	
+	Node(T data, Node<T> next){
+		this.data = data;
+		this.next = next;
+	}
+	
+	// Update the next node
+	public void nextNode(Node<T> next){
+		this.next = next;
+	}
+	
+	public T getData(){
+		return data;
+	}
+	
+	public Node<T> getNext(){
+		return next;
+	}
+}
+
+// Create a SimpleLinkedList object to hold Nodes
+class SimpleLinkedList<T>{
+	Node<T> root = null;
+	Node<T> last = null;
+	int size = 0;
+	
+	SimpleLinkedList(){}
+	
+	SimpleLinkedList(T rootData){
+		Node<T> rootNode = new Node<T>(rootData, null);
+		root = rootNode;
+		last = root;
+		size = 1;
+	}
+	
+	public int getSize(){
```

Comment:
How would you change your code if the Linked List provided didn't contain a pre-calculated size method?